# Data types in Python Notebook

##   Chapter 4 - Data types in Python.

Basic data types integer, float and string
advanced data types : list, tuple and dictionary

### 4.1 Integers

Numbers with out decimal places eg. -5 - 5
declaration variableName = initial value
    
e.g 
-3, -2 -1, 0, 1, 2, 3
e.g mobile numbers equals integers.




In [5]:
age = 20
num = 12345678
print(f'Integers : age:{age}\n Mobile number:{num}')

Integers : age:20
 Mobile number:12345678


### 4.2 Float
Numbers with decimal parts as 3.14 etc.
Declares in the same way as integers.

In [6]:
height, weight = 1.82, 67.2
print(f'floated numbers : \n height: {height}cm\n weight: {weight}kg')

floated numbers : 
 height: 1.82cm
 weight: 67.2kg


### 4.3 Strings
strings refers to  text. both single and double quotes can be used as a string. ('', "")
concatenate sign (+) can be used to combine multiple strings

Python has multiple functions to manipulate strings like **upper()**, **lower()** 

upper() method to manipulate the string to capitalize all the letters in a string.

lower() a method to manipulate the string to lower all the letters in a string.

capitalizee() a method to manipulate the string to capitalize the first letter a string.

#### Formated strings

Strings can be formated using the % operator. It gives you a grater control over how you want your string to be displayed and stored. 

The syntax for using the % operator
print('text to be formated') % (var, var1, var2)

The parentheses behind the % sign is actually known as a tuple. These placeholders will be replaced with the variable *brand*, the value *1299* & the variable *exr*

%s formatter is used to represent a string
%d formatter is used to represent an integer, can add an integer between % d to represent the length
%f formatter is used to format floats (numbers with decimals). we formated it as %4.2f (length, decimalplaces)



In [22]:
#   Initializing variables
txt = 'Hello World\n'

txt = txt.upper()
print(txt)

txt = txt.lower()
print(txt)

txt = txt.capitalize()
print(txt)

brand, exr = 'Android', 1.2345
msg = "The price of this %s phone is %d EUR and the exchange rate is %4.2f USD to 1 EUR" % (brand, 1299, exr)
print(msg)

HELLO WORLD

hello world

Hello world

The price of this Android phone is 1299 EUR and the exchange rate is 1.23 USD to 1 EUR


### 4.4 Type Casting in Python
    Converting string into integers, integers into string etc
    bultin function : **int()**, **str()**,**float()**
    int("1") converts into 1, but int("1.2") doesnt work neither does int("str")

### 4.5 List
    Refers to a collection of dataq which are normally related. They are ordered
    e.g 
        uAlias = ['jhon', 'Jake', 'Timmb']
        uAge = [21,23,24,51,53]

    Can be accsessed using index 
    e.g 
    uAlias[0]

    Notation uAlias[start:end] is known as slice, includes a third number known as stepper uAlias[start:end:Stepper]
    Slices has a default value whihch is zero

    bult-in functions :
        .append() - Add one more to the list,
        del list[index] - remove one from the list.



### 4.6 Tuples
    Declaring a tuple : tupleName = ()
    ex :
        monthsOfYear = ("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec")

    Same as list, beside it can NOT be altered when the tuple is set.



### 4.7 Dictonary
        Dictionary : dictionaryName {key:data} / dict(key:data)
                                        - keys are requered to accsess the information, keys can be strings, integers

    A collection of related data PAIRS
    Set same as dicitonary besides the key, and they're not ordered.
    acsessing a dictonary dictName[key]
    adding values to a dictonary dictName['key'] = value
    removing values from a dictonary del dictName[key]